client/signup/steps/design-picker/index.jsx

```diff
@@ -59,8 +62,19 @@ export default function DesignPickerStep( props ) {
 	const [ selectedDesign, setSelectedDesign ] = useState( null );
 	const scrollTop = useRef( 0 );
 
+	// Limit themes to those that support the Site editor, if site is fse eligible
+	const siteId = useSelector( ( state ) => getSiteId( state, dependencies.siteSlug ) );
```

Comment:
Because we're now `dependencies.siteSlug` directly, I realised we need to test a whole other path. When the user adds a custom domain during signup and goes through checkout then we don't use the `?siteSlug` query param, we use the `?siteId` query param. We have code that ensures the `signupDependencies.siteSlug` is correctly populated, and it appears to be working. But I wanted to mention it as an FYI.
If this turns out to cause a problem then the `getSelectedSiteId()` selector is another option

Reply:
Thanks for pointing this out. I just tested this with a new user, including a plan and domain purchase. I see the siteId param in the url when I get to the site setup flow and it the design picker loads the correct FSE themes without any problems that I can see.